WcaOnRails/app/models/result.rb

```diff
@@ -38,8 +39,15 @@ def unique_result_per_round
 
   scope :final, -> { where(roundTypeId: RoundType.final_rounds.map(&:id)) }
   scope :succeeded, -> { where("best > 0") }
+  scope :average_succeeded, -> { where("average > 0") }
   scope :podium, -> { final.succeeded.where(pos: [1..3]) }
   scope :winners, -> { final.succeeded.where(pos: 1).joins(:event).order("Events.rank") }
+  scope :no_later_than, lambda { |date|
+    joins(:competition).where("start_date <= ?", date)
```

Comment:
I'm not sure that applies, because that's about solves in a multi-day round, rather than a multi-day competition.

The really correct way to do this would be based on the competition schedule. That sounds like a more complicated join, though, and I'm not that enthusiastic about trying to write that join.

Maybe this is a question for WCAT? I'll start a thread with them. WCRP does not specify:

> 5.1 The competitor limit can be on either a “first come first serve” basis or based upon
qualification results or rankings in the WCA world rankings of a previously announced
calendar date.

Reply:
I changed this to `:before` and used `end_date` rather than `start_date`, and strictly-less-than.